src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -589,5 +638,95 @@ private static string GetFullPath(K8SConfiguration configuration, string path)
                 return Path.Combine(Path.GetDirectoryName(configuration.FileName), path);
             }
         }
+
+        private static void MergeKubeConfig(K8SConfiguration basek8SConfig, K8SConfiguration mergek8SConfig)
+        {
+            // For scalar values, prefer local values 
+            basek8SConfig.ApiVersion = basek8SConfig.ApiVersion ?? mergek8SConfig.ApiVersion;
+            basek8SConfig.CurrentContext = basek8SConfig.CurrentContext ?? mergek8SConfig.CurrentContext;
+            basek8SConfig.FileName = basek8SConfig.FileName ?? mergek8SConfig.FileName;
+            basek8SConfig.Kind = basek8SConfig.Kind ?? mergek8SConfig.Kind;
+
+            // For Dictionaries, prefer first if not present in second.
+            if (mergek8SConfig.Preferences != null && mergek8SConfig.Preferences.Count > 0)
+            {
+                foreach (var preference in mergek8SConfig.Preferences)
+                {
+                    if (basek8SConfig.Preferences?.ContainsKey(preference.Key) == false)
+                    {
+                        basek8SConfig.Preferences[preference.Key] = preference.Value;
+                    }
+                }
+            }
+
+            if (mergek8SConfig.Extensions != null && mergek8SConfig.Extensions.Count > 0)
+            {
+                foreach (var extension in mergek8SConfig.Extensions)
+                {
+                    if (basek8SConfig.Extensions?.ContainsKey(extension.Key) == false)
+                    {
+                        basek8SConfig.Extensions[extension.Key] = extension.Value;
+                    }
+                }
+            }
+
+            // For IEnumerable, use name as unique identifier for whether we should insert or not.
+            if (mergek8SConfig.Contexts != null && mergek8SConfig.Contexts.Count() > 0)
+            {
+                var contexts = new List<Context>();
+                if (basek8SConfig.Contexts != null)
+                {
+                    contexts.AddRange(basek8SConfig.Contexts);
```

Comment:
This is definitely going to run into problems (weird problems in kubectl b/c golang randomizes iteration) if contexts are duplicated. You can't just blindly add things into the lists, you need to make sure there are no duplicates.

Reply:
oh, oops, nevermind, I see you are checking below.